### PR TITLE
Demo Page

### LINKED ISSUE #5214: Request a demo page development
**Landing page**
🔗[Request a demo meta](https://docs.google.com/spreadsheets/d/1ZoI8zTJ91M6zrTa0MWXv8asc3fubuZV1KSOGxXRiGgE/edit#gid=1572625403) 
🔗[Figma file ](https://www.figma.com/file/VkHZfQKzUwnBEsWcRxm8ks/Demo-page?node-id=0%3A1)
↩️ Redirects from https://info.sourcegraph.com/demo-request to the new page 

Asana: https://app.asana.com/0/1200095757619647/1202008254024999/f

Hubspot form. Remember to add Chilipiper 🌶️
```
<!--[if lte IE 8]>
<script charset="utf-8" type="text/javascript" src="//js.hsforms.net/forms/v2-legacy.js"></script>
<![endif]-->
<script charset="utf-8" type="text/javascript" src="//js.hsforms.net/forms/v2.js"></script>
<script>
 hbspt.forms.create({
 region: "na1",
 portalId: "2762526",
 formId: "baf7d112-bb19-4b95-851e-a83e7b214b9b"
});
</script>
```

### PR DESCRIPTION
**Pull request by bretthayes**
- Closes #5214 and ports the Request a Demo page over from [the info subdomain](https://info.sourcegraph.com/demo-request)
- Updates all relevant links to point to the new `/demo` page

## Test
- Ensure meta and Figma match
- Ensure HubSpot form works and ChiliPiper gets triggered after submission
- Ensure all links work that were updated

### CODE REVIEW

File: website/src/pages/demo.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        <Layout
+            location={props.location}
+            meta={{
+                title: 'Request a Demo | Sourcegraph',
+                description: `From developer onboarding to incident response, see how companies of all sizes use Sourcegraph to solve the industry's most challenging code problems.`,
+            }}
+            heroAndHeaderClassName="bg-gradient-green-blue"
+            hero={
+                <div className="container pt-6 pb-4">
+                    <h1 className="font-weight-bolder">Request a demo</h1>
```
Comment (katjuell) [inline website/src/pages/demo.tsx:25]: Figma has semi-bold for this looks like

**Review by katjuell — CHANGES_REQUESTED** [review]
One teeny change & should be good to fly!

File: website/src/pages/demo.tsx (hunk 1 above)
Comment (katjuell) [reply website/src/pages/demo.tsx:25]: Actually, I see that there are two; was looking at this
Screen Shot 2022-03-29 at 9 56 12 AM

Maybe this was updated?

**Review by elzannewentzel — COMMENTED** [review]
Looks good !

File: website/src/pages/demo.tsx (hunk 1 above)
Comment (bretthayes) [reply website/src/pages/demo.tsx:25]: I believe the Figma artboard at the bottom is the correct artboard to follow which also meets feature parity with the current https://info.sourcegraph.com/demo-request page. Both use a 700 font weight.

Also, Bootstrap doesn't have a semibold font weight utility class so if we want to switch we'll have to kick off a typography sass partial.

Comment (katjuell) [reply website/src/pages/demo.tsx:25]: Our `font-weight-bold` does 600: https://github.com/sourcegraph/about/blob/4f2dba1f1057960b24ebecc78b0f1d9da81f9f43/website/src/css/styles.scss#L20-L22
Screen Shot 2022-03-29 at 1 55 07 PM

**Review by katjuell — APPROVED** [review]
lgtm!

**Comment by bretthayes** [comment]
Just need an approval from you @elzannewentzel letting me know I'm good to merge and stakeholders have approved.

File: website/src/pages/demo.tsx (hunk 1 above)
Comment (bretthayes) [reply website/src/pages/demo.tsx:25]: > Our `font-weight-bold` does 600:

[oh no lol](https://i.imgflip.com/6am5xb.jpg)

Comment (katjuell) [reply website/src/pages/demo.tsx:25]: lololololol

**Comment by elzannewentzel** [comment]
Approved!

**Pull request merged**